File: src/main/kotlin/info/hildegynoid/cue/converter/entity/CueTrackIndex.kt
package info.hildegynoid.cue.converter.entity

data class CueTrackIndex(
    var minutes: Int,
    var seconds: Int,
    var frames: Int,
    var hours: Int = 0 // rekordbox-dj
) {
    override fun toString(): String {
        val min = minutes + hours * 60
        return String.format("%02d:%02d:%02d", min, seconds, frames)
    }
}

File: src/cue/entity/CueTrackIndex.ts
import { zeroFill } from '../Utils';

export class CueTrackIndex {
  minutes: number;

  seconds: number;

  frames: number;

  hours: number | null; // for rekordbox-dj

  constructor(
    minutes: number,
    seconds: number,
    frames: number,
    hours?: number
  ) {
    this.minutes = minutes;
    this.seconds = seconds;
    this.frames = frames;
    this.hours = hours ?? null;
  }

  toString = (): string => {
    if (this.hours) {
      const min = this.minutes + this.hours * 60;
      return `${zeroFill(min)}:${zeroFill(this.seconds)}:${zeroFill(
        this.frames
      )}`;
    }
    return `${zeroFill(this.minutes)}:${zeroFill(this.seconds)}:${zeroFill(
      this.frames
    )}`;
  };
}
